include/moqt/publish_track_handler.h

```diff
@@ -0,0 +1,267 @@
+/*
+ *  Copyright (C) 2024
+ *  Cisco Systems, Inc.
+ *  All Rights Reserved
+ */
+#pragma once
+
+#include <functional>
+#include <moqt/core/base_track_handler.h>
+#include <moqt/metrics.h>
+
+namespace moq::transport {
+
+    /**
+     * @brief MOQ track handler for published track
+     *
+     * @details MOQ publish track handler defines all track related callbacks and
+     *  functions for publish. Track handler operates on a single track (namespace + name).
+     *
+     *  This extends the base track handler to add publish (aka send) handling
+     */
+    class PublishTrackHandler : protected BaseTrackHandler
+    {
+      public:
+        friend class Transport;
+
+        /**
+         * @brief Publish status codes
+         */
+        enum class PublishStatus : uint8_t
+        {
+            kOk = 0,
+            kInternalError,
+            kNotAuthorized,
+            kNotAnnounced,
+            kNoSubscribers,
+            kObjectPayloadLengthExceeded,
+            kPreviousObjectTruncated,
+
+            kNoPreviousObject,
+            kObjectDataComplete,
+            kObjectContinuationDataNeeded,
+            kObjectDataIncomplete,              ///< PublishObject() was called when continuation data remains
+
+            /// Indicates that the published object data is too large based on the object header payload size plus
+            /// any/all data that was sent already.
+            kObjectDataTooLarge,
+
+            /// Previous object payload has not been completed and new object cannot start in per-group track mode
+            /// unless new group is used.
+            kPreviousObjectNotCompleteMustStartNewGroup,
+
+            /// Previous object payload has not been completed and new object cannot start in per-track track mode
+            /// without creating a new track. This requires to unpublish and to publish track again.
+            kPreviousObjectNotCompleteMustStartNewTrack,
+        };
+
+        /**
+         * @brief  Status codes for the publish track
+         */
+        enum class Status : uint8_t
+        {
+            kOK = 0,
+            kNotConnected,
+            kNotAnnounced,
+            kPendingAnnounceResponse,
+            kAnnounceNotAuthorized,
+            kNoSubscribers,
+            kSendingUnannounce,
+        };
+
+
+        // --------------------------------------------------------------------------
+        // Public API methods that normally should not be overridden
+        // --------------------------------------------------------------------------
```

Comment:
Curios, Why is the restriction to not overridde and what issues happen if one does override ?

Reply:
It doesn't matter. I'll remove that non public comment.